File: 백준/Silver/2606. 바이러스/바이러스.py
import sys
input = sys.stdin.readline

n = int(input())
m = int(input())

graph = [[] for _ in range(n+1)]

for _ in range (m):
    s, e = map(int, input().split())
    graph[e].append(s)
    graph[s].append(e)

for i in range(n+1):
    graph[i].sort()


visit = [0] * (n+1)
visit[1] = 1

cnt = 0

def dfs(v):

    global cnt
    cnt += 1


    for nv in graph[v]:
        if visit[nv] == 0:
            visit[nv] = 1
            dfs(nv)
dfs(1)
cnt -= 1

print(cnt)

File: 백준/Silver/2606. 바이러스/바이러스.java
import java.util.*;
import java.io.*;

public class Main {
	
	static StringBuilder sb = new StringBuilder();
	static int node, line, s, e, start, cnt;
	static int[][] graph;
	static boolean[] visited;
	
	public static void main(String[] args) throws IOException{
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringTokenizer st;
		
		node = Integer.parseInt(br.readLine());
		line = Integer.parseInt(br.readLine());
		graph = new int[node+1][node+1];
		visited = new boolean[node+1];
		start = 1;
		cnt = 0;
		
		for(int i=1; i<=line; i++) {
			st = new StringTokenizer(br.readLine(), " ");
			s = Integer.parseInt(st.nextToken());
			e = Integer.parseInt(st.nextToken());
			graph[s][e] = e;
			graph[e][s] = s;
		}
		dfs(start);
		System.out.println(cnt);
	}
	public static void dfs(int s) {
		visited[s] = true;
		for (int i=1; i<=node; i++) {
			if(graph[s][i] != 0 && !visited[i]) {
				cnt ++;
				visited[i] = true;
				dfs(i);
			}
			
		}
		
		
	}
	
}